app/lib/activitypub/activity/create.rb

```diff
@@ -89,8 +89,8 @@ def process_status_params
       text: converted_object_type? ? converted_text : (@status_parser.text || ''),
       language: @status_parser.language,
       spoiler_text: converted_object_type? ? '' : (@status_parser.spoiler_text || ''),
-      created_at: @status_parser.created_at,
-      edited_at: @status_parser.edited_at && @status_parser.edited_at != @status_parser.created_at ? @status_parser.edited_at : nil,
+      created_at: [Time.now.utc, @status_parser.created_at].compact.min,
```

Comment:
This line and line below are doing same thing ... could add a private method to this class (or StatusParser?) to wrap up the "only give me past times" idea.

Separately ... given possible small clock variations ... should there be any future tolerance at all? I guess if something is reported as being 2 mins from now and we correct it to be right now instead, that's probably fine?

Also separately - add some specs for this one and the trends/statuses changes?

Reply:
> I guess if something is reported as being 2 mins from now and we correct it to be right now instead, that's probably fine?

Yes, that's my view as well. We should not respect future dates.

Reply:
I'm unsure we should override the posted date, but right now I can't see a use case for publishing in the future.

Reply:
it feels like every time the actual representation of the object is overridden locally we step further away from the possibility of making redistributable objects/activities, and if the major concern here is the placement in the discovery algo then the change should be limited to that rather than the representation of the object itself

Reply:
> it feels like every time the actual representation of the object is overridden locally we step further away from the possibility of making redistributable objects/activities, and if the major concern here is the placement in the discovery algo then the change should be limited to that rather than the representation of the object itself.

I hadn't thought about that aspect, and don't really have a strong opinion about the first part of that - but re: the latter part, it does seem like a PR which changed *just the trending calculation* (with spec showing future time issue) might solve the narrow thing here, and leave this open as draft to contemplate the more substantial date rewriting?

Reply:
Well, the problem is that if I just clamp the date used in the trend calculation to the current time, the post will still stick to the top perpetually with enough interactions.

We support backdating posts and that's totally fine but I'm not sure what would be the valid use case for future-dating posts. It shouldn't be relevant to any archive imports or account migrations.

Reply:
Maybe we need to make this kind of thing we did here with trusted/untrusted value pairs more generic, since it seems like it's a general problem and other fields like this would benefit from it: https://github.com/mastodon/mastodon/pull/32620/commits/

Reply:
What about simply excluding future posts from trending? They might start to trend when their time comes, but this probably isn't much of an issue?